x/datapool/keeper/pool_distribute.go

```diff
@@ -0,0 +1,177 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/medibloc/panacea-core/v2/types/assets"
+	"github.com/medibloc/panacea-core/v2/x/datapool/types"
+)
+
+func (k Keeper) DistributeRevenuePools(ctx sdk.Context) error {
+	revenuePools, err := k.getDistributeRevenuePools(ctx)
+	if err != nil {
+		return err
+	}
+
+	if revenuePools.IsEmpty() {
+		return nil
+	}
+
+	// This lastIndex is to execute the following iteration in the next block if it does not end in one block.
+	// But We're currently processing it all in one block. :)
+	lastIndex := 0
+	// TODO We need to think about how to deal with the pool where distribution has failed.
+	for i, poolID := range revenuePools.PoolIds {
+		lastIndex = i
+		// search current pool info
+		pool, err := k.GetPool(ctx, poolID)
+		if err != nil {
+			return err
+		}
+
+		// calculate the revenue to be sent to each seller(totalSalesBalance / totalShareTokenAmount)
+		eachDistributionAmount, err := k.getEachDistributionAmount(ctx, pool)
+		if err != nil {
+			return err
+		}
+
+		// send deposit to curator
+		err = k.sendDepositToCurator(ctx, pool)
+		if err != nil {
+			return err
+		}
+
+		// distribute of revenue to each seller
+		k.bankKeeper.IterateAllBalances(ctx, func(addr sdk.AccAddress, balance sdk.Coin) bool {
+			if balance.Denom != types.GetDenomOfShareToken(poolID) {
+				return false
+			}
+			// Seller must receive a distribution equal to the shareTokens it owns.
+			distributeCoins := sdk.NewCoins(
+				sdk.NewCoin(assets.MicroMedDenom, eachDistributionAmount.Mul(balance.Amount)),
+			)
+
+			poolAddress, err := sdk.AccAddressFromBech32(pool.GetPoolAddress())
+			// stop distribute when an error occurs
+			if err != nil {
+				return true
+			}
+
+			// excluding moduleAccount
+			if k.accountKeeper.GetModuleAddress(types.ModuleName).Equals(addr) {
+				return false
+			}
+
+			err = k.bankKeeper.SendCoins(ctx, poolAddress, addr, distributeCoins)
+
+			if err != nil {
+				k.Logger(ctx).Error("failed to distribute coin", err)
+				return true
+			}
+			return false
+		})
+
+		// an error occur when distributing revenue to sellers
+		if err != nil {
```

Comment:
This err may not be from the `IterateAllBalances()`. 
The callback, which is an argument of `IterateAllBalances()`, returns true when error occurs, but the function `IterateAllBalances()` itself doesn't return anything. How can we catch the error from the iteration?

Reply:
If any error occurs, this iterator stops running and terminates (return true).
When exiting, there is a value in err, so you can check whether an error has occurred by checking the error in this part.

Is there any part where the err value can be set other than IterateAllBalances()?

Reply:
I found the problem you mentioned and fixed it.
Thank you :)

https://github.com/medibloc/panacea-core/pull/311/commits/ac000361124256562269861ca5e2c145265de268#diff-815b7bd8fbeec0907b84af82002300154955256bf90959d4b142a0f84b75830fR54